jina/executors/indexers/__init__.py

```diff
@@ -156,7 +156,11 @@ def close(self):
 
     def flush(self):
         """Flush all buffered data to ``index_abspath`` """
-        call_obj_fn(self.write_handler, 'flush')
+        try:
+            # It may have already been closed by the Pea using context manager
+            call_obj_fn(self.write_handler, 'flush')
```

Comment:
Oh does this address https://github.com/jina-ai/jina/issues/1413 somehow?

Reply:
Could be.

Reply:
But I am not sure, because this is only at close time